src/components/BackToTopButton/BackToTopButton.svelte

```diff
@@ -0,0 +1,158 @@
+<script>
+	import { onMount } from 'svelte';
+
+	// All sizing props can be passed with or without 'px' unit.
+	// For example, '40px' and '40' are both valid values.
+	export let size = '40px';
+	export let padding = '12px';
+	export let color = '';
+	export let bgColor = '';
+	export let border = '1px';
+	export let borderRadius = '2px';
+	export let margin = '20px';
+	// The button will be visible when the user scrolls down
+	// by 'showAt' value.
+	export let showAt = 50;
+
+	// Function for auto appending 'px' unit.
+	function appendPx(value = '') {
+		value = value.toLowerCase();
+		if (
+			value.endsWith('%') ||
+			value.endsWith('vh') ||
+			value.endsWith('vw') ||
+			value.endsWith('em') ||
+			value.endsWith('rem') ||
+			value.endsWith('ex') ||
+			value.endsWith('ch') ||
+			value.endsWith('vmin') ||
+			value.endsWith('vmax') ||
+			value.endsWith('cm') ||
+			value.endsWith('mm') ||
+			value.endsWith('in') ||
+			value.endsWith('pt') ||
+			value.endsWith('pc')
+		) {
+			return value;
+		}
+		if (!value.endsWith('px')) {
+			return value + 'px';
+		}
+		return value;
+	}
+
+	// Auto append 'px' unit for all props if not provided.
+	size = appendPx(size);
+	padding = appendPx(padding);
+	border = appendPx(border);
+	borderRadius = appendPx(borderRadius);
+	margin = appendPx(margin);
+
+	let scale = 1;
+	try {
+		// Parse float using substring to remove 'px' unit.
+		scale = parseFloat(size.substring(0, size.length - 2)) / 40;
+	} catch (e) {
+		// Do nothing.
+	}
+
+	// (Default) minus value means the user has not scrolled yet.
+	// So, the button will be initially shown.
+	let scrollY = -1;
+	let show = true;
+
+	function onScroll() {
+		// Set 'scrollY' value.
+		scrollY = window.scrollY;
+
+		// Only show button when the user has scrolled down
+		// by 'showAt' value.
+		show = scrollY >= 0 && scrollY >= showAt;
+	}
+
+	function onClick() {
+		window.scrollTo({ top: 0, behavior: 'smooth' });
+	}
+
+	onMount(() => {
+		window.addEventListener('scroll', onScroll);
+
+		// Call onScroll() for  the first time.
+		onScroll();
+	});
+</script>
+
+<!-- Add keydown event & tabindex to make 'div' button accessible by keyboard -->
+<!-- Add role 'button' to make 'div' button accessible to all users -->
+<div
+	class="back-to-top-button"
+	class:hide={!show}
+	style="--size:{size}; 
+            --padding:{padding};
+            --color:{color}; 
+            --bgColor:{bgColor}; 
+            --border:{border}; 
+            --border-radius:{borderRadius};
+            --margin:{margin};
+            --scale: {scale}"
+	on:click={onClick}
+	on:keydown={onClick}
+	tabindex="0"
+	role="button"
```

Comment:
I think using `<button>` element is more straightforward since we don't need to add these properties for i18n and `curser: pointer`.

Reply:
I think we already add "CSS reset" to handle those case krub (Both Carbon and Tailwind have this feature). Using a button element here should have the same style as a normal div. (If not then please let me know lol)

Reply:
My bad regarding this assumption. The carbon CSS that we use actually does not reset the native button style. So I turned it off and used Tailwind reset instead. Now it's working as expected.